[bugfix] check122 has to check not only string but also array values of the Action field

kagahd opened the pull request:
### Context

The `check122` tested only `string` but not `array` values of the `Action` field. Therefore `check122` did not find IAM policies where a star `*` value of the `Action` field is an `array` element. 

Example of a policy that `check122` did **not** find **before** the bugfix:
```
{
    "Statement": [
        {
            "Action": [
                "*"
            ],
            "Effect": "Allow",
            "Resource": "*"
        }
    ],
    "Version": "2012-10-17"
}
```
**Before** this bugfix, `check122` did find IAM policies only like this:
```
{
    "Statement": [
        {
            "Action": "*",
            "Effect": "Allow",
            "Resource": "*",
            "Sid": ""
        }
    ],
    "Version": "2012-10-17"
}
```

### Description

I changed the code `Action == '*'`, which tests only `string` values, to `contains(Action, '*')`, which now also tests `array` values. Therefore `check122` will now find both types of the above shown IAM policy examples.


### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

n4ch04 commented:
Hi @kagahd that issue is already solved in Prowler v3, we encourage you to try it. 
It adds dozens of new checks, amazing performance improvements and a lot of fancy features

kagahd commented:
Thanks @n4ch04 for your quick reply.

> is already solved in Prowler v3

Yes, I know, because my team is in the process of migrating from prowler v2 to v3. The [official prowler v2/v3 mapping](https://github.com/prowler-cloud/prowler/blob/master/docs/tutorials/aws/v2_to_v3_checks_mapping.md) maps v2 `check122` to v3 check `iam_policy_no_administrative_privileges`. However, that's wrong (may be I should open a further PR for this) because `iam_policy_no_administrative_privileges` checks for "open" IAM **AWS managed** policies whereas `check122` checks for IAM **custom** policies (see `--scope Local` in the `check122` code).
It seems that v3 has 2 checks that correspond to v2 `check122`:
- `iam_customer_attached_policy_no_administrative_privileges` and 
- `iam_customer_unattached_policy_no_administrative_privilege`

However we wondered why both v3 checks had more findings than v2 `check122` on all our more than 300 AWS accounts.
That's why I've investigated the cause and fixed `check122`. Once fixed, `check122` reported exactly the same issues as both v3 checks `iam_customer_attached_policy_no_administrative_privileges`and `iam_customer_unattached_policy_no_administrative_privilege`.
Since I just fixed `check122` and prowler v2 is still officially supported and maintained, I thought I share my work with the community.

n4ch04 commented:
Thanks for your contribution @kagahd !!
You are right, we split that check into two to allow Prowler to be more accurate, but we did not update the docs.
I'll review v3 PR to add that mapping to close this.

n4ch04 commented:
Also, to be updated about all the new fancy features we are adding to the tool we recommend you and your team to join the Prowler slack. You will be also in touch with our amazing community.
I'll leave here the link https://join.slack.com/t/prowler-workspace/shared_invite/zt-22du63ply-sXxJQP_b0ejfKeS_jZY7yA
